lldb/source/Plugins/LanguageRuntime/CPlusPlus/CPPLanguageRuntime.cpp

```diff
@@ -90,9 +79,26 @@ class LibCXXFrameRecognizer : public StackFrameRecognizer {
     if (!sc.function)
       return {};
 
+    // Check if we have a regex match
+    bool matches_regex = false;
     for (RegularExpression &r : m_hidden_regex)
-      if (r.Execute(sc.function->GetNameNoArguments()))
+      if (r.Execute(sc.function->GetNameNoArguments())) {
+        matches_regex = true;
+        break;
+      }
+
+    if (matches_regex) {
```

Comment:
Why split this out of the loop above?

Reply:
purely stylistic. I find the code easier to read if every section takes care of a single thing.

In this case, I find it more readable if we have two separate code blocks:
1. first the code block that check if we match the regular expression
2. then a separate code block which checks for the parent frame

But this is clearly a subjective argument. If you want me to write this in some other way, please let me know 🙂

Reply:
Personally I prefer not having that extra `matches_regex` variable. The first time I was reading through it I was trying to figure out if there was more to this loop that I was missing. So if you don't mind changing it to:
```
   for (RegularExpression &r : m_hidden_regex) {
      if (!r.Execute(sc.function->GetNameNoArguments()))
        continue;

      // remaining checks
    }
```
that would be appreciated